nodelet/pkg/utils/container_runtime/containerd.go

```diff
@@ -0,0 +1,316 @@
+package containerruntime
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/oci"
+	"github.com/pkg/errors"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/constants"
+	"go.uber.org/zap"
+
+	"time"
+
+	"github.com/containerd/containerd/errdefs"
+)
+
+type Containerd struct {
+	Service string
+	Socket  string
+	Client  *containerd.Client
+	log     *zap.SugaredLogger
+}
+
+func NewContainerd() (ContainerUtils, error) {
+
+	containerdclient, err := NewContainerdClient()
+	if err != nil {
+		return nil, err
+	}
+	return &Containerd{
+		Service: constants.RuntimeContainerd,
+		Socket:  constants.ContainerdSocket,
+		Client:  containerdclient,
+		log:     zap.S(),
+	}, nil
+}
+
+func NewContainerdClient() (*containerd.Client, error) {
+
+	containerdclient, err := containerd.New(constants.ContainerdSocket)
+	if err != nil {
+		zap.S().Info("failed to create containerd client")
+		return nil, errors.Wrap(err, "failed to create containerd client")
+	}
+	return containerdclient, nil
+}
+
+func (c *Containerd) EnsureFreshContainerRunning(ctx context.Context, containerName string, containerImage string) error {
+
+	err := c.EnsureContainerDestroyed(ctx, containerName, "10s")
```

Comment:
move "10s" string to constants.go or define it as a `const` at the top of this file